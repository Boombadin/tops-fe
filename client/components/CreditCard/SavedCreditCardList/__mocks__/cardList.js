export default [
  {
    bankIcon: '/images/payment/product-img-empty.svg',
    bankId: null,
    bankImage: '/images/payment/product-img-empty.svg',
    bankName: null,
    expiryDate: '12/33',
    expiryMonth: 12,
    expiryYear: 2033,
    id: '590925219668099073',
    isDefault: true,
    isExpired: false,
    maskedNumber: '411111XXXXXX1111',
    maskedNumberByStar: '**** **** **** 1111',
    type: 'Visa',
    typeIcon: '/icons/types-credit-card/visa-no-border.svg',
  },
  {
    bankIcon: '/images/payment/product-img-empty.svg',
    bankId: '10',
    bankImage: '/images/payment/product-img-empty.svg',
    bankName: 'KCC',
    expiryDate: '12/33',
    expiryMonth: 12,
    expiryYear: 2033,
    id: '604578118707118081',
    isDefault: false,
    isExpired: false,
    maskedNumber: '512345XXXXXX2346',
    maskedNumberByStar: '**** **** **** 2346',
    type: 'MasterCard',
    typeIcon: '/icons/types-credit-card/master-no-border.svg',
  },
];
