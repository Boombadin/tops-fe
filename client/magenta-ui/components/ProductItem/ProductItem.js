import { ceil, get, isEmpty, last, noop } from 'lodash';
import moment from 'moment';
import PropTypes from 'prop-types';
import React from 'react';
import { withRouter } from 'react-router-dom';
import { Icon } from 'semantic-ui-react';

import {
  ProductAddAttribute,
  ProductAddAttributeNoQty,
  ProductImpressionAttr,
} from '@client/features/gtm/models/Product';
import CartActionButton from '@client/features/product/components/CartActionButton';
import Image from '@client/magenta-ui/components/Image';
import RibbonItem from '@client/magenta-ui/components/RibbonItem';
import { GTM_CLASSES } from '@client/magenta-ui/constants/googleTagManager';
import { formatPrice } from '@client/magenta-ui/utils/price';
import { eventTracking } from '@client/utils/datalakeTracking';

import './ProductItem.scss';

export const ProductItem = React.memo(
  ({
    product,
    className,
    id,
    sku,
    img,
    title,
    priceLabel,
    price,
    specialPrice,
    url,
    onAddToWishlist,
    isAddedToWishlist,
    onRemoveFromWishlist,
    disableAddToCart,
    addToCartLabel,
    outOfStockLabel,
    qty, // qty in cart,
    promoIconsNode,
    promoNamesNode,
    promoLinkNode,
    saveLabel,
    unitLabel,
    type,
    notifyTooltipNode,
    greyOutNotAvailable,
    productBadge,
    baseMediaUrl,
    orderItem,
    trackingSection,
    trackingUserId,
    section,
    history,
    beforeAddProductCallback,
  }) => {
    const addToWishlist = e => {
      e.preventDefault();
      e.stopPropagation();

      isAddedToWishlist ? onRemoveFromWishlist(id) : onAddToWishlist(id);
    };
    const gtmAddtoCart = ProductAddAttribute(product, section);
    const gtmProductClick = ProductAddAttributeNoQty(product);
    const productImpressionsAttr = ProductImpressionAttr(
      product,
      section,
      orderItem,
    );

    let discount = null;
    let showPrice = null;
    let isInRange = false;

    if (get(product, 'extension_attributes.promotion.end_date')) {
      const currentTime = moment().format('YYYY-MM-DD HH:mm');
      isInRange = get(product, 'extension_attributes.promotion.end_date', '')
        ? currentTime <=
          moment(
            get(product, 'extension_attributes.promotion.end_date', ''),
            '',
          )
            .add(25200000 - 36000, 'ms')
            .format('YYYY-MM-DD HH:mm')
        : true;
    } else if (specialPrice && specialPrice > 0) {
      const currentTime = moment().format('YYYY-MM-DD HH:mm');
      isInRange = get(product, 'special_to_date', '')
        ? currentTime <=
          moment(get(product, 'special_to_date', ''), '')
            .add(25200000 - 36000, 'ms')
            .format('YYYY-MM-DD HH:mm')
        : true;
    }

    const promoType = get(product, 'extension_attributes.promotion.type', '')
      .toLowerCase()
      .replace(' ', '');

    if (
      specialPrice &&
      price &&
      price - specialPrice !== 0 &&
      isInRange &&
      promoType &&
      (promoType === 'redhot' || promoType === 'sale')
    ) {
      discount = (
        <div className="ribbon">
          <RibbonItem
            title={`${saveLabel} ${ceil(price - specialPrice, 2)} ${unitLabel}`}
          />
        </div>
      );
    }

    if (specialPrice && specialPrice != price && isInRange) {
      showPrice = (
        <div className="price-elem">
          <p className="mt-price-line">{price}</p>
          <p>
            <span className="mt-price-number">
              {formatPrice(Number(specialPrice))}{' '}
            </span>
            <span className="mt-price-label">
              {' '}
              /{!isEmpty(priceLabel) && last(priceLabel.split(' '))}
            </span>
          </p>
        </div>
      );
    } else {
      showPrice = (
        <div className="price-elem">
          <p>
            <span className="mt-price-number">{formatPrice(price)} </span>
            <span className="mt-price-label">
              {' '}
              /{!isEmpty(priceLabel) && last(priceLabel.split(' '))}
            </span>
          </p>
        </div>
      );
    }

    // GTM
    const addDataLayer = () => {
      if (trackingSection !== '') {
        eventTracking(
          'product_click',
          trackingUserId,
          sku,
          orderItem,
          url,
          price,
          trackingSection,
        );
      }
    };
    const handleClickProduct = () => {
      history.push(url);
    };
    switch (type) {
      case 'mini':
        return (
          <div className="mt-product-item mt-product-mini">
            <div className="product-item">
              <div className="product-item--image">
                <img src={img} alt={`Tops online ${title}`} width="200" />
              </div>
              <div className="product-detail-container">
                <div className="product-item--title">
                  <h3>{title}</h3>
                </div>
                <div className="product-item--price-wrapper">
                  <div className="product-item--price">{showPrice}</div>
                  <div className="product-item--seasonal-badge">
                    {get(productBadge, 'image_path') && (
                      <img
                        className="product-item-badge"
                        src={`${baseMediaUrl}${get(
                          productBadge,
                          'image_path',
                          '',
                        )}`}
                      />
                    )}
                  </div>
                </div>
              </div>
              <div className="product-item--button">
                {notifyTooltipNode}
                <CartActionButton
                  addToCartLabel={addToCartLabel}
                  product={product}
                  gtm={gtmAddtoCart}
                  outOfStockLabel={outOfStockLabel}
                  beforeAddProductCallback={beforeAddProductCallback}
                  type={type}
                />
              </div>
            </div>
          </div>
        );
      default:
        return (
          <div
            {...gtmProductClick}
            className={`mt-product-item ${className} ${GTM_CLASSES.TRACK_PRODUCT_CLICK}`}
            onClick={addDataLayer}
          >
            {greyOutNotAvailable && disableAddToCart && (
              <div className="mt-product-item__grey-out" />
            )}
            <div className={'product-item'}>
              <Icon
                className={`mt-add-to-wishlist${
                  isAddedToWishlist ? ' active' : ''
                }`}
                name="like"
                size="large"
                onClick={addToWishlist}
              />
              <div className="product-item--promo-icon mt-promo-icons">
                {promoIconsNode}
              </div>
              <div
                className="product-item-inner-wrap"
                onClick={handleClickProduct}
              >
                <div className="product-item--image">
                  <Image
                    src={img}
                    alt={`Tops online ${title}`}
                    width="200"
                    sku={sku}
                    orderItem={orderItem}
                    product={product}
                    section={section}
                  />
                </div>
                <div className="product-item--ribbon">{discount}</div>
                <div className="product-item--title">
                  <h3 {...productImpressionsAttr}>{title}</h3>
                </div>
                <div className="product-item--promo-name mt-promo-names">
                  {promoNamesNode}
                </div>
              </div>
              <div className="product-item--promo-link">{promoLinkNode}</div>
              <div className="product-item--price-wrapper">
                <div className="product-item--price">{showPrice}</div>
                <div className="product-item--seasonal-badge">
                  {get(productBadge, 'image_path') && (
                    <img
                      className="product-item-badge"
                      src={`${baseMediaUrl}${get(
                        productBadge,
                        'image_path',
                        '',
                      )}`}
                      title={get(productBadge, 'label', '')}
                    />
                  )}
                </div>
              </div>
              <div
                className="product-item--button"
                onClick={event => {
                  event.preventDefault();
                  event.stopPropagation();
                }}
              >
                {notifyTooltipNode}
                <CartActionButton
                  addToCartLabel={addToCartLabel}
                  product={product}
                  gtm={gtmAddtoCart}
                  outOfStockLabel={outOfStockLabel}
                  beforeAddProductCallback={beforeAddProductCallback}
                />
              </div>
            </div>
          </div>
        );
    }
  },
);

ProductItem.propTypes = {
  className: PropTypes.string,
  img: PropTypes.string,
  price: PropTypes.node.isRequired,
  specialPrice: PropTypes.number.isRequired,
  priceLabel: PropTypes.string,
  addToCartLabel: PropTypes.string,
  outOfStockLabel: PropTypes.string,
  qty: PropTypes.number,
  promoNamesNode: PropTypes.node,
  promoIconsNode: PropTypes.node,
  loadingCartProduct: PropTypes.bool,
  productBadge: PropTypes.object,
  baseMediaUrl: PropTypes.string,
  trackingSection: PropTypes.string,
  trackingUserId: PropTypes.string,
  productType: PropTypes.string,
  beforeAddProductCallback: PropTypes.func,
};

ProductItem.defaultProps = {
  className: '',
  img: 'http://via.placeholder.com/350x250',
  promoNamesNode: null,
  promoIconsNode: null,
  priceLabel: 'ชิ้น',
  addToCartLabel: 'Add to Cart',
  outOfStockLabel: 'Out of Stock',
  loadingCartProduct: false,
  productBadge: {},
  baseMediaUrl: '',
  trackingSection: '',
  trackingUserId: '',
  productType: '',
  beforeAddProductCallback: noop,
};

export default withRouter(ProductItem);
