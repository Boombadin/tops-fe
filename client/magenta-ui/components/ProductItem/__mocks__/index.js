export * from '@client/magenta-ui/components/ProductItem/__mocks__/product';
