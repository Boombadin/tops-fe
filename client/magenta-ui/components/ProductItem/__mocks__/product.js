export const mockProduct = JSON.parse(
  '{"id":47427,"type_id":"simple","name":"ฟาร์มเฮ้าส์ขนมปังโฮลวีต 500กรัม","brand_name":"1860","small_image":"/8/8/8850123110436.jpg","image":"/8/8/8850123110436.jpg","thumbnail":"/8/8/8850123110436.jpg","url_key":"farmhouse-wholewheat-bread-500g-8850123110436","country_of_product":"5457","attribute_set_id":4,"visibility":"4","status":"1","recommended":"1","promotion_no":[],"promotion_type":[],"is_returnable":"2","category":[{"category_id":2,"name":"Default Category","level":1,"parent_id":1,"url_key":"default-category","url_path":""},{"category_id":2753,"name":"อาหารสดและเบเกอรี่","level":2,"parent_id":2,"url_key":"fresh-food-bakery","url_path":"fresh-food-bakery"},{"category_id":2754,"name":"เบเกอรี่","level":3,"parent_id":2753,"url_key":"bakery","url_path":"fresh-food-bakery/bakery"},{"category_id":2958,"name":"ขนมปังแผ่น","level":4,"parent_id":2754,"url_key":"sliced-bread","url_path":"fresh-food-bakery/bakery/sliced-bread"},{"category_id":3358,"name":"แคมเปญ","level":2,"parent_id":2,"url_key":"campaign","url_path":"campaign"},{"category_id":3922,"name":"Corporate","level":2,"parent_id":2,"url_key":"corporate","url_path":"corporate"},{"category_id":3856,"name":"Catalog","level":2,"parent_id":2,"url_key":"catalog","url_path":"catalog"},{"category_id":11486,"name":"Farmhouse","level":3,"parent_id":3856,"url_key":"farmhouse","url_path":"catalog/farmhouse"},{"category_id":11487,"name":"Bakery","level":4,"parent_id":11486,"url_key":"bakery","url_path":"catalog/farmhouse/bakery"}],"price":40,"tier_prices":[{"price":40,"original_price":40,"customer_group_id":0,"is_discount":false},{"price":40,"original_price":40,"customer_group_id":1,"is_discount":false},{"price":40,"original_price":40,"customer_group_id":2,"is_discount":false},{"price":40,"original_price":40,"customer_group_id":3,"is_discount":false},{"price":40,"original_price":40,"customer_group_id":4,"is_discount":false},{"price":40,"original_price":40,"customer_group_id":5,"is_discount":false},{"price":40,"original_price":40,"customer_group_id":6,"is_discount":false}],"special_price":40,"stock":{"qty":28,"is_in_stock":true},"custom_attributes_option":[{"attribute_code":"small_image","value":"/8/8/8850123110436.jpg","position":0},{"attribute_code":"promotion_no","position":0},{"attribute_code":"price_discount","value":0,"position":0},{"attribute_code":"required_options","value":false,"position":0},{"attribute_code":"country_of_product","value":"ฝรั่งเศส","position":0},{"attribute_code":"status","value":"Enabled","position":0},{"attribute_code":"brand_code","value":"00723","position":0},{"attribute_code":"created_at","value":"2018-04-02 08:09:53","position":0},{"attribute_code":"item_type_indicator","value":"Credit Item","position":0},{"attribute_code":"manual_boosting","value":134,"position":0},{"attribute_code":"updated_at","value":"2019-08-23 10:45:14","position":0},{"attribute_code":"options_container","value":"Block after Info Column","position":0},{"attribute_code":"tax_class_id","value":"Taxable Goods","position":0},{"attribute_code":"has_options","value":false,"position":0},{"attribute_code":"url_key","value":"farmhouse-wholewheat-bread-500g-8850123110436","values":["farmhouse-wholewheat-bread-500g-8850123110436"],"position":0},{"attribute_code":"ranking","value":0.0197,"position":0},{"attribute_code":"item_remark","value":"Cut and Cooking Method Remark","position":0},{"attribute_code":"price_saving","value":0,"position":0},{"attribute_code":"promotion_end","position":0},{"attribute_code":"is_returnable","value":"2","position":0},{"attribute_code":"special_expires_at","value":"2019-11-29 05:07:31","position":0},{"attribute_code":"manual_disable","value":"Manual Disable","position":0},{"attribute_code":"image","value":"/8/8/8850123110436.jpg","position":0},{"attribute_code":"promotion_type","value":[],"position":0},{"attribute_code":"thumbnail","value":"/8/8/8850123110436.jpg","position":0},{"attribute_code":"visibility","value":4,"position":0},{"attribute_code":"product_ingredient","value":"<p>แป้งสาลี 100%</p>","position":0},{"attribute_code":"package_uom","value":"PGRA","position":0},{"attribute_code":"brand_name","value":"ฟาร์มเฮ้าส์","position":0},{"attribute_code":"meta_keyword","value":"ขนมปัง  ขนมปังแผ่น FARMHOUSE","position":0},{"attribute_code":"local_import","value":"Local","position":0},{"attribute_code":"consumer_unit","value":"ชิ้น","position":0},{"attribute_code":"search_keywords","value":"ฟาร์มเฮ้าส์ขนมปังโฮลวีต 500กรัม Farmhouse Wholewheat Bread 500g.","position":0},{"attribute_code":"product_property","value":"<p>ฟาร์มเฮ้าส์ขนมปังโฮลวีตขนมปังโฮลวีต ใช้แป้งสาลีไม่ฟอกขาว 100% ปราศจากคอลเลสเตอรอล เป็นแหล่งของเหล็ก ไนอะซิน มีวิตามินบี 1 สูง มีวิตามินบี 2 สูง และมีโฟเลตสูง</p>","position":0},{"attribute_code":"product_usage","value":"<p>ใช้รับประทานเป็นของว่างหรือทานเล่น</p>","position":0},{"attribute_code":"recommended","value":"Recommended","position":0}],"specification_attributes":[{"attribute_code":"brand_name","label":"Brand"},{"attribute_code":"product_property","label":"Product property"},{"attribute_code":"product_ingredient","label":"Product ingredient"},{"attribute_code":"product_usage","label":"Product usage"},{"attribute_code":"lifestyle_and_benefit","label":"Lifestyle and Benefit"},{"attribute_code":"sold_qty","label":"Product Sold by Website"},{"attribute_code":"is_new_arrivals","label":"Is New Arrivals"},{"attribute_code":"selling_unit","label":"selling_unit"}],"price_discount":0,"price_saving":0,"media_gallery_entries":[{"id":47427,"media_type":"image","position":1,"disabled":false,"types":["image","small_image","thumbnail"],"file":"/8/8/8850123110436.jpg"}],"breadcrumbs":[{"category_id":2753,"name":"อาหารสดและเบเกอรี่","level":1,"url":"fresh-food-bakery"},{"category_id":2754,"name":"เบเกอรี่","level":2,"url":"fresh-food-bakery/bakery"},{"category_id":2958,"name":"ขนมปังแผ่น","level":3,"url":"fresh-food-bakery/bakery/sliced-bread"}],"ranking":"0.019700","star_rating_one":0,"star_rating_two":0,"star_rating_three":0,"star_rating_four":0,"star_rating_five":0,"extension_attributes":{"stock_item":{"item_id":2459969,"product_id":47427,"stock_id":9,"qty":28,"min_qty":0,"use_config_min_qty":true,"is_qty_decimal":false,"backorders":0,"use_config_backorders":true,"min_sale_qty":1,"use_config_min_sale_qty":1,"max_sale_qty":24,"use_config_max_sale_qty":false,"is_in_stock":true,"low_stock_date":null,"notify_stock_qty":0,"use_config_notify_stock_qty":true,"manage_stock":true,"use_config_manage_stock":true,"stock_status_changed_auto":0,"qty_increments":1,"use_config_qty_increments":true,"enable_qty_increments":false,"use_config_enable_qty_increments":true,"is_decimal_divided":false,"show_default_notification_message":false},"gtm_data":{"brand_en":"FARMHOUSE","product_name_en":"Farmhouse Wholewheat Bread 500g.","category_en":"2958>Sliced Bread","category_id_level_two":2754},"t1c_redeemable_points":[320],"t1c_earn_points_estimate":[320],"website_ids":[0,2,3,4,5,9,23,26,27,34,44,46,49,52,53,67]},"created_at":"2018-04-02 08:09:53","updated_at":"2019-08-23 10:45:14","sku":"8850123110436","qty":0,"isAddedToWishlist":false,"options_container":"container2","gift_message_available":"0","gift_wrapping_available":"0","brand_code":"00723","search_keywords":"ฟาร์มเฮ้าส์ขนมปังโฮลวีต 500กรัม Farmhouse Wholewheat Bread 500g.","meta_keyword":"ขนมปัง  ขนมปังแผ่น FARMHOUSE","product_property":"<p>ฟาร์มเฮ้าส์ขนมปังโฮลวีตขนมปังโฮลวีต ใช้แป้งสาลีไม่ฟอกขาว 100% ปราศจากคอลเลสเตอรอล เป็นแหล่งของเหล็ก ไนอะซิน มีวิตามินบี 1 สูง มีวิตามินบี 2 สูง และมีโฟเลตสูง</p>","product_ingredient":"<p>แป้งสาลี 100%</p>","product_usage":"<p>ใช้รับประทานเป็นของว่างหรือทานเล่น</p>","special_from_date":"","special_to_date":"","special_expires_at":"2019-11-29 05:07:31","tax_class_id":"2","consumer_unit":"5481","exclusive_flag":"0","trending":"0","promotion_exclude":"0","is_alcohol_restriction":"0","item_remark":"5244","is_weight_scale":"0","package_uom":"5478","catch_weight_ind":"0","pack_ind":"0","simple_pack_ind":"0","local_import":"5394","item_type_indicator":"5425","alcohol_code":"0","is_nyb":"0","manual_boosting":"134","manual_disable":"1","weight_item_ind":"0","url":"/farmhouse-wholewheat-bread-500g-8850123110436"}',
);
