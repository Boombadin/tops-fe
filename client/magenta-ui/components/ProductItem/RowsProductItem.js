import { get, last, noop } from 'lodash';
import moment from 'moment';
import PropTypes from 'prop-types';
import React from 'react';
import LazyLoad from 'react-lazyload';
import { NavLink } from 'react-router-dom';
import { Icon } from 'semantic-ui-react';

import {
  ProductAddAttribute,
  ProductAddAttributeNoQty,
  ProductImpressionAttr,
} from '@client/features/gtm/models/Product';
import CartActionButton from '@client/features/product/components/CartActionButton';
import Image from '@client/magenta-ui/components/Image';
import { GTM_CLASSES } from '@client/magenta-ui/constants/googleTagManager';
import { formatPrice } from '@client/magenta-ui/utils/price';

import './RowsProductItem.scss';
import './ProductItem.scss';

export const ProductItem = React.memo(
  ({
    product,
    className,
    id,
    sku,
    img,
    title,
    priceLabel,
    price,
    specialPrice,
    url,
    onAddToWishlist,
    onRemoveFromWishlist,
    isAddedToWishlist,
    disableAddToCart,
    addToCartLabel,
    outOfStockLabel,
    promoIconsNode,
    promoNamesNode,
    promoLinkNode,
    greyOutNotAvailable,
    orderItem,
    section,
    beforeAddProductCallback,
  }) => {
    const addToWishlist = e => {
      e.preventDefault();
      e.stopPropagation();
      isAddedToWishlist ? onRemoveFromWishlist(id) : onAddToWishlist(id);
    };
    const gtmAddtoCart = ProductAddAttribute(product, section);
    const gtmProductClick = ProductAddAttributeNoQty(product);

    let showPrice = null;
    let isInRange = false;
    if (get(product, 'extension_attributes.promotion.end_date')) {
      const currentTime = moment().format('YYYY-MM-DD HH:mm');
      isInRange = get(product, 'extension_attributes.promotion.end_date', '')
        ? currentTime <=
          moment(
            get(product, 'extension_attributes.promotion.end_date', ''),
            '',
          )
            .add(25200000 - 36000, 'ms')
            .format('YYYY-MM-DD HH:mm')
        : true;
    }

    if (isInRange && specialPrice && specialPrice != price) {
      showPrice = (
        <div className="price-elem">
          <p className="mt-price-line">{price}</p>
          <p>
            <span className="mt-price-number">
              {formatPrice(Number(specialPrice))}{' '}
            </span>
            <span className="mt-price-label">
              {' '}
              /{last(priceLabel.split(' '))}
            </span>
          </p>
        </div>
      );
    } else {
      showPrice = (
        <div className="price-elem">
          <p>
            <span className="mt-price-number">{formatPrice(price)} </span>
            <span className="mt-price-label">
              {' '}
              /{last(priceLabel.split(' '))}
            </span>
          </p>
        </div>
      );
    }
    const productImpressionsAttr = ProductImpressionAttr(
      product,
      section,
      orderItem,
    );
    return (
      <NavLink
        {...gtmProductClick}
        to={url}
        className={`mt-mt-pi-rows ${className} ${GTM_CLASSES.TRACK_PRODUCT_CLICK}`}
      >
        <div className="mt-pi-rows">
          {greyOutNotAvailable && disableAddToCart && (
            <div className="mt-mt-pi-rows__grey-out" />
          )}
          <div className="mt-pi-rows--image-column">
            <div className="mt-pi-rows--image">
              <LazyLoad height={100}>
                <Image
                  src={img}
                  sku={sku}
                  orderItem={orderItem}
                  product={product}
                  section={section}
                />
              </LazyLoad>
            </div>
          </div>
          <div className="mt-pi-rows--content-column">
            <div className="mt-pi-rows--title" {...productImpressionsAttr}>
              {title}
            </div>
            <div className="mt-pi-rows--promo-name mt-promo-names rows-layout">
              {promoNamesNode}
            </div>
            <div className="mt-pi-rows--promo-link">{promoLinkNode}</div>
            <div className="mt-pi-rows--promo-icon mt-promo-icons rows-layout">
              {promoIconsNode}
            </div>
            <div className="mt-pi-rows--price">{showPrice}</div>
          </div>
          <div className="mt-pi-rows--controls-column">
            <div className="mt-pi-rows--button">
              <CartActionButton
                addToCartLabel={addToCartLabel}
                product={product}
                gtm={gtmAddtoCart}
                outOfStockLabel={outOfStockLabel}
                beforeAddProductCallback={beforeAddProductCallback}
              />
            </div>
            <Icon
              className={`mt-add-to-wishlist--rows${
                isAddedToWishlist ? ' active' : ''
              }`}
              name="like"
              size="large"
              onClick={addToWishlist}
            />
          </div>
        </div>
      </NavLink>
    );
  },
);

ProductItem.propTypes = {
  isRowMode: PropTypes.bool,
  className: PropTypes.string,
  img: PropTypes.string,
  price: PropTypes.node,
  spacialPrice: PropTypes.node,
  priceLabel: PropTypes.string,
  addToCartLabel: PropTypes.string,
  outOfStockLabel: PropTypes.string,
  qty: PropTypes.number,
  promoNamesNode: PropTypes.node,
  promoIconsNode: PropTypes.node,
  badge: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      imgPromo: PropTypes.string.isRequired,
      type: PropTypes.string.isRequired,
    }),
  ),
  url: PropTypes.string,
  discount: PropTypes.string,
  beforeAddProductCallback: PropTypes.func,
};

ProductItem.defaultProps = {
  isRowMode: false,
  className: '',
  img: 'http://via.placeholder.com/350x250',
  price: '',
  spacialPrice: '',
  url: '#',
  promoNamesNode: null,
  promoIconsNode: null,
  badge: [],
  discount: '',
  priceLabel: 'ชิ้น',
  addToCartLabel: 'Add to Cart',
  outOfStockLabel: 'Out of Stock',
  beforeAddProductCallback: noop,
};

export default ProductItem;
