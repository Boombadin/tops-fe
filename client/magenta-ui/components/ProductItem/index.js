import React from 'react';

import ProductItem from '@client/magenta-ui/components/ProductItem/ProductItem';
import RowsProductItem from '@client/magenta-ui/components/ProductItem/RowsProductItem';

export default props =>
  props.isRowMode ? <RowsProductItem {...props} /> : <ProductItem {...props} />;
