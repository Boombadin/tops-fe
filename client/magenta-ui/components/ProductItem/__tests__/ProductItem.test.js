import { cleanup, render } from '@testing-library/react'; // import library
import React from 'react';

import { mockProduct } from '@client/magenta-ui/components/ProductItem/__mocks__';
import { ProductItem } from '@client/magenta-ui/components/ProductItem/ProductItem';

jest.mock('@client/contexts', () => ({
  useCartContext: jest.fn().mockReturnValue({ cartAction: {} }),
  useReduxContext: jest.fn().mockReturnValue({ reduxAction: {} }),
}));

afterEach(cleanup); // will remove param or props each describe (unmount container render)

describe('ProductItem', () => {
  const { container } = render(
    <ProductItem
      product={mockProduct}
      addToCartLabel="ใส่รถเข็น"
      baseMediaUrl="https://staging23-mdc.tops.co.th/media/"
      className="swiper-slide"
      disableAddToCart={false}
      id={47427}
      img="https://staging23-mdc.tops.co.th/media/catalog/product/8/8/8850123110436.jpg"
      isAddedToWishlist={false}
      loadingCartProduct={false}
      maxQty={24}
      onChangeQty={() => {}}
      orderItem={29}
      outOfStockLabel="ขออภัยสินค้าหมด"
      price={40}
      priceLabel="ชิ้น"
      qty={0}
      reduxState={{}}
      reduxDispatch={() => {}}
      saveLabel="ลด"
      section="Recommended for you"
      sku="8850123110436"
      specialPrice={40}
      stock={28}
      title="ฟาร์มเฮ้าส์ขนมปังโฮลวีต 500กรัม"
      trackingSection="homepage_recommendation"
      trackingUserId="541436"
      type="slider"
      unitLabel="บาท"
      url="/farmhouse-wholewheat-bread-500g-8850123110436"
    />,
  );

  test('it correctly renders the component', () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
