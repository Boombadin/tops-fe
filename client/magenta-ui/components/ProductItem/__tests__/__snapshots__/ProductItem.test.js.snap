// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem it correctly renders the component 1`] = `
<div
  class="mt-product-item swiper-slide Js-trackProductClick"
  data-productbrand="FARMHOUSE"
  data-productcategory="2958>Sliced Bread"
  data-productid="8850123110436"
  data-productname="Farmhouse Wholewheat Bread 500g."
  data-productprice="40"
>
  <div
    class="product-item"
  >
    <i
      aria-hidden="true"
      class="like large icon mt-add-to-wishlist"
    />
    <div
      class="product-item--promo-icon mt-promo-icons"
    />
    <div
      class="product-item-inner-wrap"
    >
      <div
        class="product-item--image"
      >
        <img
          alt="Tops online ฟาร์มเฮ้าส์ขนมปังโฮลวีต 500กรัม"
          class="lazyload undefined mt-image loading Image__Img-sc-1s8ofg8-0 hULgNN"
          data-pid="8850123110436"
          data-position="30"
          data-productbrand="FARMHOUSE"
          data-productcategory="2958>Sliced Bread"
          data-productname="Farmhouse Wholewheat Bread 500g."
          data-productoriginalprice="40"
          data-productprice="40"
          data-productsection="Recommended for you"
          data-src="https://staging23-mdc.tops.co.th/media/catalog/product/8/8/8850123110436.jpg"
          width="auto"
        />
      </div>
      <div
        class="product-item--ribbon"
      />
      <div
        class="product-item--title"
      >
        <h3
          data-pid="8850123110436"
          data-position="30"
          data-productbrand="FARMHOUSE"
          data-productcategory="2958>Sliced Bread"
          data-productname="Farmhouse Wholewheat Bread 500g."
          data-productoriginalprice="40"
          data-productprice="40"
          data-productsection="Recommended for you"
        >
          ฟาร์มเฮ้าส์ขนมปังโฮลวีต 500กรัม
        </h3>
      </div>
      <div
        class="product-item--promo-name mt-promo-names"
      />
    </div>
    <div
      class="product-item--promo-link"
    />
    <div
      class="product-item--price-wrapper"
    >
      <div
        class="product-item--price"
      >
        <div
          class="price-elem"
        >
          <p>
            <span
              class="mt-price-number"
            >
              40.00
               
            </span>
            <span
              class="mt-price-label"
            >
               
              /
              ชิ้น
            </span>
          </p>
        </div>
      </div>
      <div
        class="product-item--seasonal-badge"
      />
    </div>
    <div
      class="product-item--button"
    >
      <button
        class="AddProductButton__Button-ag50oj-0 bkWnZn"
        data-productbrand="FARMHOUSE"
        data-productcategory="2958>Sliced Bread"
        data-productid="8850123110436"
        data-productname="Farmhouse Wholewheat Bread 500g."
        data-productprice="40"
        data-productsection="Recommended for you"
        data-qty="1"
        data-testid="add-product-button"
        type="default"
      >
        ใส่รถเข็น
      </button>
    </div>
  </div>
</div>
`;
